spec/models/transaction_spec.rb

```diff
@@ -0,0 +1,13 @@
+require "rails_helper"
+
+RSpec.describe Transaction do 
+  describe "relationships" do 
+    it { should belong_to :invoice }
+  end
+
+  describe "validations" do 
+    it { should validate_presence_of :credit_card_number }
+    it { should validate_presence_of :credit_card_expiration_date }
+    it { should validate_presence_of :result }
```

Comment:
Same for here with the enums, something like

it { should define_enum_for(:status).with_values([success: 0, failed: 1]) }